src/platform/tasks/vscode/tasksService.ts

```diff
@@ -19,14 +19,29 @@ import { IWorkspaceService } from '../../workspace/common/workspaceService';
 import { ITasksService, TaskResult, TaskStatus } from '../common/tasksService';
 
 
-export class TasksService implements ITasksService {
+export class TasksService extends DisposableStore implements ITasksService {
 	_serviceBrand: undefined;
 
+	private taskDefinitionToTerminal: Map<vscode.TaskDefinition, vscode.Terminal> = new Map();
 	constructor(
 		@IWorkspaceService private readonly workspaceService: IWorkspaceService,
 		@IFileSystemService private readonly fileSystemService: IFileSystemService,
 		@ILanguageDiagnosticsService private readonly languageDiagnosticsService: ILanguageDiagnosticsService,
-	) { }
+	) {
+		super();
+		this.add(vscode.tasks.onDidStartTask(e => {
+			const terminal: vscode.Terminal | undefined = (e.execution as any).terminal;
+			if (!terminal) {
+				return;
+			}
+			this.taskDefinitionToTerminal.set(e.execution.task.definition, terminal);
+			this.add(vscode.window.onDidCloseTerminal(closedTerminal => {
```

Comment:
This leaks a listener every time a task gets started. Should make sure to clear this listener when appropriate